components/camel-spring-ws/src/main/java/org/apache/camel/component/spring/ws/SpringWebserviceConsumer.java

```diff
@@ -57,21 +57,25 @@ public void invoke(MessageContext messageContext) throws Exception {
 
         // start message processing
         getProcessor().process(exchange);
-
+        Message responseMessage = null;
+        Source responseBody = null;
         if (exchange.getException() != null) {
             throw exchange.getException();
         } else if (exchange.getPattern().isOutCapable()) {
-            Message responseMessage = exchange.getOut(Message.class);
-            if (responseMessage != null) {
-                Source responseBody = responseMessage.getBody(Source.class);
-                WebServiceMessage response = messageContext.getResponse();
+            responseMessage = exchange.getOut(Message.class);
+        } else {
+            responseMessage = exchange.getIn(Message.class);
```

Comment:
I thoughy i should do hasOut() check here as well, then i thought exchange.getPattern().isOutCapable() is no use. If we check hasOut() then we can use getOut or getIn relatively and accordingly. that's because i am not sure if we really need to check the out capability.